### ISSUE TITLE
rez must not be null

### ISSUE THREAD

**Issue by stefanos-kalantzis** [EMResearch/EvoMaster#239]
I'm trying to use EvoMaster on my project.
I followed this guide: https://github.com/EMResearch/EvoMaster/blob/master/docs/write_driver.md

After a lot of issues with P6Spy, I managed to run everything, but I get this exception:

```
* Starting to generate test cases
* [ERROR] EvoMaster process terminated abruptly. This is likely a bug in EvoMaster. Please copy&paste the following stacktrace, and create a new issue on https://github.com/EMResearch/EvoMaster/issues
java.lang.IllegalStateException: rez must not be null
        at org.evomaster.core.output.oracles.SchemaOracle$getSupportedResponse$1.accept(SchemaOracle.kt:215)
        at org.evomaster.core.output.oracles.SchemaOracle$getSupportedResponse$1.accept(SchemaOracle.kt:32)
        at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:684)
        at org.evomaster.core.output.oracles.SchemaOracle.getSupportedResponse(SchemaOracle.kt:182)
        at org.evomaster.core.output.oracles.SchemaOracle.generatesExpectation(SchemaOracle.kt:230)
        at org.evomaster.core.output.PartialOracles.activeOracles(PartialOracles.kt:164)
        at org.evomaster.core.problem.rest.service.AbstractRestFitness.handleResponseTargets(AbstractRestFitness.kt:366)
        at org.evomaster.core.problem.rest.service.RestFitness.doCalculateCoverage(RestFitness.kt:112)
        at org.evomaster.core.problem.rest.service.RestFitness.doCalculateCoverage(RestFitness.kt:22)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage(FitnessFunction.kt:59)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage$default(FitnessFunction.kt:46)
        at org.evomaster.core.search.algorithms.MioAlgorithm.search(MioAlgorithm.kt:38)
        at org.evomaster.core.Main$Companion.run(Main.kt:284)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:150)
        at org.evomaster.core.Main$Companion.main(Main.kt:72)
        at org.evomaster.core.Main.main(Main.kt)
```

_(unfortunately P6Spy seems to only work with `h2` on my project. it would be nice to run it on my production database, but P6Spy don't seem to support mariadb)_

**Comment by stefanos-kalantzis** [comment]
I took a look at the code, I think the problem is here: https://github.com/EMResearch/EvoMaster/blob/master/core/src/main/kotlin/org/evomaster/core/output/oracles/SchemaOracle.kt#L212

So if none of the above lines match the object type, then it will always go here `is Schema -> valueSchema.$ref` and never here `else -> ""`.

Then, when the code goes here: `valueSchema.$ref`, there is no null-check for `$ref`.

---

_As a side note, this is a very nice example of why I think Kotlin didn't solve the java-null problem... :(_

**Comment by stefanos-kalantzis** [comment]
By changing

```
is Schema -> valueSchema.`$ref`
else -> ""
```

to

```
else -> valueSchema.`$ref` ?: ""
```

it seems to not trip anymore. but it tripped somewhere else. I'll create a separate issue :)

**Comment by arcuri82** [comment]
hi!
  thanks for trying out EvoMaster ;)
Regarding your first issue, it looks like the same as in #237
if you build jar from repository, that code should not be executed (but we haven't confirmed yet, as it does not happen on the APIs we use for debugging)

Regarding P6Spy on MariaDB, we haven't tried MariaDB yet. But it is definitely something we should handle in the future. 
Anyway, even if P6Spy is not working, it is not the end of the world. EvoMaster will perform slightly worse, as it is just that it will not analyze the SQL commands done on the DB

@bogdanmarculescu  can you double-check Stefanos's hypothesis about the bug in the code?

**Comment by bogdanmarculescu** [comment]
The hypothesis of what the underlying bug is matches mine: rez was not intended to be null, but is in some cases. 

The test cases for that oracle are not yet complete, and this is the reason I'm rather reluctant to simply change that code (why I deactivated it rather than just add the null check). Once the tests are done (And I'm confident changes don't break anything else somewhere) I'll add the more permanent fix. 

Incidentally, what else trips up if you add the null check?

**Comment by arcuri82** [comment]
this should now be fixed in the new 1.2.0 release. if not, please open a new issue
